packages/jest-matchers/src/types.js

```diff
@@ -22,6 +24,10 @@ export type RawMatcherFn = (
 
 export type ThrowingMatcherFn = (actual: any) => void;
 export type MatcherContext = {isNot: boolean};
+export type MatcherState = {
+  currentTestName?: string,
+  currentTestPath?: Path,
```

Comment:
since every test file has its own instance of matchers, it would be more appropriate to call it `testPath` and to drop the `current`.